indexing-service/src/main/java/org/apache/druid/indexing/common/IndexTaskClient.java

```diff
@@ -262,6 +276,7 @@ protected StringFullResponseHolder submitSmileRequest(
         encodedPathSuffix,
         encodedQueryString,
         content,
+        new StringFullResponseHandler(StandardCharsets.UTF_8),
```

Comment:
with this change, I believe the following code paths are no longer reachable, since the response would always be a success
- https://github.com/apache/druid/blob/2df0f52cfad28c3be2f43fb57b9ca3415c6f3eea/indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTaskClient.java#L66-L71
- https://github.com/apache/druid/blob/2df0f52cfad28c3be2f43fb57b9ca3415c6f3eea/indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTaskClient.java#L101-L106
- https://github.com/apache/druid/blob/2df0f52cfad28c3be2f43fb57b9ca3415c6f3eea/indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTaskClient.java#L129-L133

The calls to `IndexTaskClient.isSuccess(StringFullResponseHolder)` from `SeekableStreamIndexTaskClient` would also no longer be necessary.

As a result I believe we can remove the `IndexTaskClient.isSucess()` method entirely